source/features/select-notifications.tsx

```diff
@@ -0,0 +1,176 @@
+						break;
+					}
+
+					continue;
+				}
+
+				if (categoryFilters.length > 0 && !select.exists(categoryFilters, notification)) {
+					isNotificationSelected = false;
+					break;
+				}
+			}
+
+			// Make excluded notifications unselectable
+			if (!isNotificationSelected) {
+				select('.js-notification-bulk-action-check-item', notification)!.removeAttribute('data-check-all-item');
+			}
+		}
+
+		// If at least one notification is selectable, trigger the "Select all" checkbox
+		if (select.exists('.js-notification-bulk-action-check-item[data-check-all-item]')) {
+			selectAllCheckbox.dispatchEvent(new MouseEvent('click'));
+		}
+	}
+
+	// Make all notifications selectable again
+	for (const disabledNotificationCheckbox of select.all('.js-notification-bulk-action-check-item:not([data-check-all-item])')) {
+		// eslint-disable-next-line unicorn/prefer-dom-node-dataset -- For consistency with the `removeAttribute()` line
+		disabledNotificationCheckbox.setAttribute('data-check-all-item', '');
+	}
+}
+
+function createDropdownList(category: Category, filters: Filter[]): JSX.Element {
+	const icons: {[Key in Filter]: JSX.Element} = {
+		'Pull requests': <GitPullRequestIcon className="color-text-secondary"/>,
+		Issues: <IssueOpenedIcon className="color-text-secondary"/>,
+		Open: <CheckCircleIcon className="color-text-success"/>,
+		Closed: <XCircleIcon className="color-text-danger"/>,
+		Draft: <GitPullRequestDraftIcon className="color-text-tertiary"/>,
+		Merged: <GitMergeIcon className="text-purple"/>,
+		Read: <DotIcon className="color-text-link"/>,
+		Unread: <DotFillIcon className="color-text-link"/>
+	};
+
+	return (
+		<div className="SelectMenu-list">
+			<header className="SelectMenu-header">
+				<span className="SelectMenu-title">{category}</span>
+			</header>
+			{filters.map(filter => (
+				<label
+					className="SelectMenu-item text-normal"
+					role="menuitemcheckbox"
+					aria-checked="false"
+					tabIndex={0}
+					data-filter={filter}
+					data-category={category}
+				>
+					<CheckIcon className="octicon octicon-check SelectMenu-icon SelectMenu-icon--check mr-2" aria-hidden="true"/>
+					<div className="SelectMenu-item-text">
+						<input hidden type="checkbox"/>
+						{icons[filter]}
+						<span className="ml-2">{filter}</span>
+					</div>
+				</label>
+			))}
+		</div>
+	);
+}
+
+function createDropdown(): JSX.Element {
+	return (
+		<details
+			className="details-reset details-overlay position-relative"
+			on-toggle={resetFilters}
+		>
+			<summary
+				className="btn btn-sm ml-3 mr-1"
+				data-hotkey="S"
+				aria-haspopup="menu"
+				role="button"
+			>
+				Select by <span className="dropdown-caret ml-1"/>
+			</summary>
+			<details-menu
+				className="SelectMenu left-0"
+				aria-label="Select by"
+				role="menu"
+				on-details-menu-selected={handleSelection}
+			>
+				<div className="SelectMenu-modal">
+					{createDropdownList('Type', ['Pull requests', 'Issues'])}
```

Comment:
Do we need to support discussions too?

Reply:
I don't know, I've never used them. Do they generate a lot of noise?

There are more types we could support too, like Commits, CI results, Security warnings... I think we can leave it like this and see what people think. If they ask support for more categories it'll be easy enough to add.